ext/dcos-installer/dcos_installer/backend.py

```diff
@@ -55,6 +58,55 @@ def do_aws_cf_configure():
     configure.do_aws_cf_configure(stringify_configuration(yaml_config))
 
 
+def do_upload_to_s3(config_path=CONFIG_PATH):
+    """Best effort to upload generated CF templates to S3"""
+    config = DCOSConfig(config_path=config_path)
+    must_contain_keys = [
+        'cloudformation_s3_url',
+        'cloudformation_s3_bucket',
+        'cloudformation_s3_key',
+        'cloudformation_s3_region']
+
+    for key in must_contain_keys:
+        if key not in config:
+            log.error("You must set the key for {} in the config.yaml to upload the templates.".format(key))
+            return 1
+
+    s3_bucket = config['cloudformation_s3_bucket']
+    s3_key = config['cloudformation_s3_key']
+    s3_region = config['cloudformation_s3_region']
+
+    credential_env = ['AWS_SECRET_ACCESS_KEY', 'AWS_ACCESS_KEY_ID']
+    for e in credential_env:
+        if e not in os.environ:
+            log.error("Must set {} to upload templates to S3".format(e))
+            return 1
+
+    cf_template_dir = '/genconf/serve/cloudformation'
+    if not os.path.exists(cf_template_dir):
+        log.error("genconf/serve/cloudformation does not exist. Try --aws-cloudformation.")
+        return 1
+
+    cf_files = glob.glob('{}/*'.format(cf_template_dir))
```

Comment:
We know what files we want to upload exactly, lets upload those exactly. Guessing / globbing picks up random stuff in the directory.

Also note we need to upload things like all of the packages output in the entire serve directory.